src/actions/lockDomainMetadata.ts

```diff
@@ -0,0 +1,21 @@
+import { ethers } from "ethers";
+import { Registrar } from "../contracts/types";
+import { validateOwner, validateStatus } from "./helpers";
+
+export const lockDomainMetadata = async (
+  domainId: string,
+  lockStatus: boolean,
+  potentialOwner: string,
+  registrar: Registrar
+): Promise<ethers.ContractTransaction> => {
+  validateOwner(
+    domainId,
+    potentialOwner,
+    registrar,
+    "Cannot set metadata of unowned domain"
```

Comment:
Nit: This could be interpreted as the domain is not created, or is not owned by anyone. Maybe `Must own domain to lock metadata` is more clear. Also this error message is a bit misleading since this function is locking metadata, not setting it.